### PR TITLE
[BACK-2895] Cleanup unused indexes, add missing indexes

### PR DESCRIPTION
**Pull request by Roukoswarf**

### CODE REVIEW

**Review by lostlevels — DISMISSED** [review]

File: data/store/mongo/mongo_data_set.go [hunk 1]
```diff
@@ -68,25 +66,33 @@ func (d *DataSetRepository) EnsureIndexes() error {
...
 				{Key: "_userId", Value: 1},
 				{Key: "deviceId", Value: 1},
 				{Key: "type", Value: 1},
-				{Key: "_active", Value: 1},
-				{Key: "_deduplicator.hash", Value: 1},
+				{Key: "createdTime", Value: -1},
 			},
 			Options: options.Index().
-				SetBackground(true).
+				SetName("ListUserDataSetsAlso").
```
Comment (lostlevels) [inline data/store/mongo/mongo_data_set.go:92]: Kind of confused about this name, but otherwise looks fine.

**Review by lostlevels — APPROVED** [review]

File: data/store/mongo/mongo_data_set.go (hunk 1 above)
Comment (Roukoswarf) [reply data/store/mongo/mongo_data_set.go:92]: been a while, but i believe both are used by listing user datasets in different ways, suggestions?

Comment (lostlevels) [reply data/store/mongo/mongo_data_set.go:92]: Perhaps `ListUserDataSetsByDeviceId` or `ListUserDataSetsDeviceId` or `DataSetsByDeviceId` ? Naming is hard...

Comment (Roukoswarf) [reply data/store/mongo/mongo_data_set.go]: replaced with option 2

**Review by lostlevels — APPROVED** [review]

File: data/store/mongo/mongo_test.go [hunk 2]
```diff
@@ -576,7 +344,54 @@ var _ = Describe("Mongo", func() {
...
-			It("returns successfully", func() {
+			It("deviceDataSets indexes return successfully", func() {
+				cursor, err := dataSetCollection.Indexes().List(context.Background())
+				Expect(err).ToNot(HaveOccurred())
+				Expect(cursor).ToNot(BeNil())
+				var indexes []storeStructuredMongoTest.MongoIndex
+				err = cursor.All(context.Background(), &indexes)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(err).ToNot(HaveOccurred())
```
Comment (lostlevels) [inline data/store/mongo/mongo_test.go:355]: Duplicate?

**Review by lostlevels — APPROVED** [review]

File: data/store/mongo/mongo_test.go (hunk 2 above)
Comment (Roukoswarf) [reply data/store/mongo/mongo_test.go]: yep, fixed.

**Review by lostlevels — APPROVED** [review]

**Pull request merged**